scout/server/blueprints/cases/templates/cases/utils.html

```diff
@@ -13,12 +13,25 @@
             <div class="row align-items-between align-items-center">
               <div class="col-10">
                 <i class="fa fa-check-circle-o"></i>
-		            {% if variant.category in ("snv", "cancer", "str") %}
+		            {% if variant.category in "str" %}
                   <a href="{{ url_for('variant.variant',
+                                   institute_id=variant.institute,
+                                   case_name=case.display_name,
+                                   variant_id=variant._id, str='yes') }}">
+                  {{ variant.str_repid }} {{ variant.alternative }}
+                {% elif variant.category in ("snv", "cancer") %}
+                  {% if variant.category in "cancer" %}
+                    <a href="{{ url_for('variant.variant',
                                       institute_id=variant.institute,
                                       case_name=case.display_name,
-                                      variant_id=variant._id) }}">
-                    {% set display_genes = [] %}
+                                      variant_id=variant._id, cancer='yes') }}">
```

Comment:
What about putting the conditional only in the last part, so you could avoid repetition of code. Something like:

```
{% if variant.category in ("snv", "cancer", "str") %}

<a href="{{ url_for('variant.variant',
                                   institute_id=variant.institute,
                                   case_name=case.display_name,
                                   variant_id=variant._id
{% if  variant.category ==  "str"%}
, str='yes'
{% elif variant.category ==  "cancer" %}
, cancer='yes'
{% endif %}
)}}">
{% set display_genes = [] %}
```

Reply:
One would love to, right. Only the situation is a bit complex. :wink: The var passing is already in a jinja2 statement, and we don't do like two passes of jinja, so no nested use is allowed.

If it is hard to see at a glance, try thinking about if you could close the `{{}}` block early, say after `variant._id` above. But jinja evaluation would still be trying to execute the url_for, but needing all the arguments to be in place for that.

To complicate things, the offending variable parameter here becomes the function call argument name, so I have no use of the variable directly either in the jinja/python env. If the construction was like `url_for(variant_page_type=variant.category` that would be fine. I considered getting at it using e.g. `eval()`, but I don't think the linting will love it - very soon this will look like a good old perl program.

What I would suggest is to rewrite the case and variant pages, utils and views, and dynamically determine what page to display using `variant.category` instead of passing keywords like `cancer=yes`. This is obviously tricky for the variants page, and here I would suggest changing to passing the variant type as the argument rather than argument name.

Reply:
I rant too much and write too little code; I'll check if it would work!